nautobot_golden_config/nornir_plays/config_compliance.py

```diff
@@ -45,6 +47,47 @@ def get_rules():
     return rules
 
 
+def get_config_element(rule, config, obj, logger):
+    """
+    Helper function to yield elements of the configuration as defined in the `config_match` under ComplianceRule.
+
+    Returns:
+       - a configuration section for `CLI` based config types
+       - top level JSON key for `JSON` based config types
+    """
+    if rule["obj"].config_type == ComplianceRuleTypeChoice.TYPE_JSON:
+        config_json = get_json_config(config)
+
+        if not config_json:
+            logger.log_failure(obj, "Unable to interpret configuration as JSON.")
+            raise NornirNautobotException("Unable to interpret configuration as JSON.")
+
+        if rule["obj"].config_to_match:
+            config_element = {
+                k: config_json.get(k) for k in rule["obj"].config_to_match.splitlines() if k in config_json
+            }
+        else:
+            config_element = config_json
+
+    elif rule["obj"].config_type == ComplianceRuleTypeChoice.TYPE_CLI:
+        if get_platform(obj.platform.slug) not in parser_map.keys():
+            logger.log_failure(
+                obj,
+                f"There is currently no CLI-config parser support for platform slug `{get_platform(obj.platform.slug)}`, preemptively failed.",
+            )
+            raise NornirNautobotException(
+                f"There is currently no CLI-config parser support for platform slug `{get_platform(obj.platform.slug)}`, preemptively failed."
+            )
+
+        config_element = section_config(rule, config, get_platform(obj.platform.slug))
+
+    else:
```

Comment:
Isn't this an unreachable block? There is only two ComplianceruleTypechoices and its a required field on the rule definition

Reply:
Technically yes - however , it's just a sanity check - if one adds new TYPE in the future , `get_config_element` would have to be updated